### PR TITLE
update go version

### PR DESCRIPTION
**Pull request by Avni-Sharma**
update go version to 1.14

### CODE REVIEW

**Comment by sbose78** [comment]
Do we have a 1.14 builder?

**Comment by Avni-Sharma** [comment]
> Do we have a 1.14 builder?
> https://github.com/redhat-developer/service-binding-operator/blob/master/Dockerfile.rhel#L4

Not as of now :/

**Comment by sbose78** [comment]
:(
/hold

**Comment by baijum** [comment]
@Avni-Sharma There is Go 1.14.1 version available.

**Comment by Avni-Sharma** [comment]
https://github.com/openshift/release/tree/master/projects/origin-release 1.14 is not available yet :/

File: openshift-ci/Dockerfile.tools [hunk 1]
```diff
@@ -28,7 +28,7 @@ WORKDIR /tmp
 RUN mkdir -p $GOPATH/bin
 RUN mkdir -p /tmp/goroot
 
-RUN curl -Lo go1.13.7.linux-amd64.tar.gz https://dl.google.com/go/go1.13.7.linux-amd64.tar.gz && tar -C /tmp/goroot -xzf go1.13.7.linux-amd64.tar.gz
+RUN curl -Lo go1.14.linux-amd64.tar.gz https://dl.google.com/go/go1.14.linux-amd64.tar.gz && tar -C /tmp/goroot -xzf go1.14.linux-amd64.tar.gz
```
Comment (baijum) [inline openshift-ci/Dockerfile.tools:31]: Use Go 1.14.3 (latest bugfix release).
https://dl.google.com/go/go1.14.3.linux-amd64.tar.gz

**Comment by pedjak** [comment]
should not we update `go.mod` as well?

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/redhat-developer/service-binding-operator
 
-go 1.13
+go 1.14.3
```
Comment (pedjak) [inline go.mod:3]: we should not be so precise here, 1.14 should work.

Comment (DhritiShikhar) [reply go.mod:3]: @pedjak 

Hi, Just to understand, Is there a technical reason why we don't provide the patch number?

Comment (pedjak) [reply go.mod:3]: this here is about go compiler version, not the bugfix version. It is signal to compiler what language features will be available:

https://blog.ksub.org/bytes/2019/09/15/go.mods-go-directive/

Comment (DhritiShikhar) [reply go.mod:3]: Oh thanks for explanation!

**Pull request merged**